### PR TITLE
Distributions examples for Documentation Purposes

### PR DESCRIPTION
**Pull request by stevereyes01**
An updated PR that includes using pycbc distributions.py to make distributions for mass and spins.

Note that this code will have to change as distributions.py moves from inference to wherever it goes next.

### CODE REVIEW

**Comment by stevereyes01** [comment]
Ignore codacy complaining about the import of `from mpl_toolkits.mplot3d import Axes3D`, it's definitely needed to make the 3d plot of points uniformly on a sphere. Otherwise I think this should be fine to merge (I'll update the commits when `distributions.py` changes locations).

File: examples/distributions/mass_examples.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+# Take 100000 random variable samples from this gaussian mass distribution.
+mass2_samples = mass2_distribution.rvs(size=1000000)
+
+# We can make pairs of distributions together, instead of apart.
+two_mass_distributions = distributions.Uniform(mass3=(1.6, 3.0),
+                                               mass4=(1.6, 3.0))
+two_mass_samples = two_mass_distributions.rvs(size=1000000)
+
+# Choose 20 mass bins for the histogram subplots.
```
Comment (cmbiwer) [inline examples/distributions/mass_examples.py]: I know what you mean here but I wouldn't use the phrase "mass bins" that has had a different meaning in the past. Just say histogram bins.

File: examples/distributions/mass_examples.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+n_bins = 20
+
+# Make some subplots
+fig, axes = plt.subplots(nrows=2, ncols=2)
+ax0, ax1, ax2, ax3, = axes.flat
+
+ax0.hist(mass1_samples['mass1'], bins = n_bins)
```
Comment (cmbiwer) [inline examples/distributions/mass_examples.py]: It probably would've been good to plot the PDF over the samples like:
```
import matplotlib.pyplot as plt
import numpy
from pycbc.inference.distributions import Gaussian

# set random seed
numpy.random.seed(0)

# draw samples from Gaussian distribution
g = Gaussian(mass2=(0,10), mass2_mean=5, mass2_var=1)
samples = g.rvs(size=500000)

# Get PDF
mass2 = numpy.arange(0, 10, 0.1)
pdf = [g.pdf(mass2=i) for i in mass2]

# plot drawn samples and PDF
nbins = 50
plt.hist(samples["mass2"], nbins, normed=True, histtype="step")
plt.plot(mass2, pdf)
plt.ylabel("Probability Density Function")
plt.xlabel(r"Mass 2 ($M_{\odot}$)")
plt.show()
```
Though I can add that in a future PR.

**Comment by cmbiwer** [comment]
Hey Steven can you just change the mass bins thing. Then I'll merge. Thanks for doing this! There's some things I want to add on top of the work you've done but this is a great start!

**Comment by stevereyes01** [comment]
Sure, I'll make those changes, thanks for pointing them out!

**Comment by cmbiwer** [comment]
Thanks for changing the comment Steven. I think this is a good set of examples and later we can add the PDF lines and maybe even some other examples like getting a CDF from the PDF. I'll merge when the checks finish (modulo the complaint about the ``Axes3D`` that is required by matplotlib for the 3-D plot).

**Pull request merged**